src/adapter/src/optimize/peek.rs

```diff
@@ -49,7 +49,7 @@ pub struct Optimizer {
     /// A snapshot of the cluster that will run the dataflows.
     compute_instance: ComputeInstanceSnapshot,
     /// Optional row-set finishing to be applied to the final result.
-    finishing: RowSetFinishing,
+    pub finishing: RowSetFinishing,
```

Comment:
It's nice to have this private so nobody can muck with the optimizer's internal state and potentially break its assumptions. Can we leave this non-pub and use `optimizer.finishing().clone()` instead? Or, if we are worried about the cost of cloning, have an fn into_finishing(self) -> RowSetFinishing`?

Reply:
Makes sense, backed out that change and reverted to `optimizer.finishing().clone()`.